### PR TITLE
Fix tests that were mocking ILogger

### PR DESCRIPTION
**Pull request by pakrym**
Use `TestSink.MessageLogged` instead.

### CODE REVIEW

**Comment by Tratcher** [comment]
Do you have a summary for why it was broken?

**Comment by pakrym** [comment]
We started using structs as TState in ILogger.Log<TState>().

I wasn't able to find an easy way of setting up Mock method where T is an unknown struct but it doesn't matter much because mocks are not required there at all.

File: test/Kestrel.Transport.FunctionalTests/RequestTests.cs [hunk 1]
```diff
@@ -280,37 +269,26 @@ public async Task ConnectionResetBetweenRequestsIsLoggedAsDebug()
...
-                    }
+                    return;
+                }
 
-                    if (logLevel > LogLevel.Debug)
-                    {
-                        loggedHigherThanDebug = true;
-                    }
-                });
+                if (context.LogLevel > LogLevel.Debug)
```
[the remark this answers is not in the document]
Comment (pakrym) [reply test/Kestrel.Transport.FunctionalTests/RequestTests.cs]: A very theoretical race. ` Assert.False(loggedHigherThanDebug);` may be executed earlier then loggedHigherThanDebug  is set if we call `connectionReset.Release(); ` before setting local.

**Comment by halter73** [comment]
https://github.com/moq/moq4/issues/343

**Pull request merged**